### DISCUSSION TITLE
Use salt-cloud to easily perform aws queries

### DISCUSSION THREAD

**Discussion by Akm0d** [saltstack/salt#67572] [Feature Requests]
This came up when the following command needed to be built into a pillar for cloud testing:

`salt['cmd.run']("aws ec2 describe-images --filter 'Name=name,Values=Windows_Server-2012-R2_RTM-English-64Bit-Base-*' --region us-west-2 --owner xxxxxxxxx --query 'sort_by(Images, &CreationDate)[-1].ImageId'")`

It would be nice for a salt-cloud to wrap and simplify that aws command.

**Comment by Ch3LL** [comment]
to clarify - add the ability to define an ami based on multiple filters in your cloud profile. and have it return the correct image.
